### DISCUSSION TITLE
Is there a way to add class labels to PyTrackObjects?

### DISCUSSION THREAD

**Discussion by cedarwarman** [quantumjot/btrack#215] [Q&A] [answered]
Hi, I am using BayesianTracker to track pollen tube growth and pollen grains in microscopy images. It's working great, but I am wondering if there's a way to add a class label to a PyTrackObject? For example, I have a series of bounding boxes that I'm converting to PyTrackObjects using the from_dict() method. For pollen, the bounding boxes can have classes like "ungerminated", "germinated", and "burst". When the tracking happens the classes are ignored, which is the desired behavior for me. However, once I have the track for a specific pollen grain, I'd like to be able to see what classes it had over time (e.g. at what timepoint did the class change from ungerminated to germinated or from germinated to burst). Can the "features" parameter be used for this purpose? Thanks for your help!

**Comment by quantumjot** [comment, marked as answer]
When you create the `PyTrackObject` using `from_dict()` you can add extra properties in there. For example:

<img width="811" alt="Screenshot 2023-03-09 at 06 54 12" src="https://user-images.githubusercontent.com/8217795/224015755-cf3de5d6-0a82-4b47-818a-8fb77d6996c3.png">

Then, when you track with these objects, the extra metadata should appear in the properties dictionary for each resulting track:
<img width="878" alt="Screenshot 2023-03-09 at 06 54 37" src="https://user-images.githubusercontent.com/8217795/224015921-86c425f1-90c0-401e-b0c8-c89dea4c206b.png">

**Reply by cedarwarman** [reply]
This works, thank you! I had to add the `replace_nan=False` flag to the `btrack.BayesianTrack()` method `to_napari()` for the plotting to work, e.g.:
```
data, properties, graph = tracker.to_napari(replace_nan=False)
```
